src/components/FileUpload.tsx

```diff
@@ -54,16 +55,105 @@ export default function FileUpload() {
         setErrorOpen(false);
     };
 
+    async function getToken() {
+        const token = await firebase.auth().currentUser?.getIdToken();
+
+        if (token === undefined) return {};
+
+        return token;
+    }
+
     useEffect(() => {
         globalThis.uploadedDatasets = datasets;
     }, [datasets]);
 
+    async function uploadDatatset(key: string) {
+        if (key === undefined) {
+            enqueueSnackbar("Something went wrong", {
+                variant: "error"
+            });
+            return;
+        }
+
+        fetch(process.env.NEXT_PUBLIC_GQL_API_UPDATED, {
```

Comment:
What's `process.env.NEXT_PUBLIC_GQL_API_UPDATED`? Also you should use the existing apollo client and graphql codegen for this so that things are consistent, let me know if you want me to show you how to do this. It's basically just creating an `addDataset.gql` file in `src/graphql` that contains the query, running `yarn gql-codegen`, and then importing the generated function/types from `src/generated/queries.ts`.

Reply:
`process.env.NEXT_PUBLIC_GQL_API_UPDATED` is an updated graphql api with dataset integrated, I didn't want want to remove update with the new one I thought it can break stuffs